### PR TITLE
Add commands for managing Bucket CORS

### PR DESCRIPTION
**Pull request by marktheunissen**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

Add the following commands:

- `mc cors set`
- `mc cors remove`
- `mc cors get`

## Motivation and Context

These will work with S3 and MinEOS

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/cors-set.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Set the CORS configuration for the bucket 'mybucket':
+     {{.Prompt}} {{.HelpName}} myminio/mybucket /path/to/cors.xml
+ `,
+}
+
+// corsMessage container for output message.
+type corsMessage struct {
```
Comment (zveinn) [inline cmd/cors-set.go:57]: Since this msg is used by all commands.. maybe it and it's methods should live in the cord-main.go file.

**Review by zveinn — COMMENTED** [review]
Looks solid.

File: cmd/cors-set.go (hunk 1 above)
Comment (marktheunissen) [reply cmd/cors-set.go:57]: @zveinn Good point, and I agree, but I put corsMessage in here to be consistent with the other commands that share a message struct, here are three examples:

- https://github.com/minio/mc/blob/master/cmd/admin-user-add.go#L93-L102
- https://github.com/minio/mc/blob/master/cmd/admin-group-add.go#L66-L74
- https://github.com/minio/mc/blob/master/cmd/ilm-tier-add.go#L346-L356

I am totally happy to move it, just give me the 👍 and ack that it makes it inconsistent with the rest of the commands. 

Could also make a follow up PR to move them all. I don't know why it's like this.

File: cmd/cors-set.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	console.SetColor("CorsMessage", color.New(color.FgGreen))
+
+	// args[0] is the ALIAS/BUCKET argument.
+	args := ctx.Args()
+	urlStr := args.Get(0)
+
+	// args[1] is the CORSFILE which is a local file.
+	corsXML, e := os.ReadFile(args.Get(1))
+	fatalIf(probe.NewError(e).Trace(args...), "Unable to read bucket CORS configuration file.")
```
Comment (klauspost) [inline cmd/cors-set.go:107]: Maybe accept stdin?

```suggestion
	var e error
	in := os.Stdin
	if f := args.Get(1); f != "-" {
		in, e = os.Open(f)
		fatalIf(probe.NewError(e).Trace(args...), "Unable to read bucket CORS configuration file.")
	}
	corsXML, e := io.ReadAll(in)
	fatalIf(probe.NewError(e).Trace(args...), "Unable to read bucket CORS configuration file.")
	in.Close()
```

Comment (marktheunissen) [reply cmd/cors-set.go]: Thanks, good suggestion, have pushed a new commit with this change

**Review by klauspost — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**